marker_uilints/tests/ui/print_span.stderr

```diff
@@ -1,33 +1,50 @@
 warning: print span
  --> $DIR/print_span.rs:9:5
   |
-9 |     let _span_normal = 178;
-  |     ^^^^^^^^^^^^^^^^^^^^^^^
+9 |     let _span_try = Some(21)?;
+  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
-  = note: Debug: Span {
-              source: File(
-                  "$DIR/print_span.rs",
-              ),
-              start: 134,
-              end: 137,
-          }
-  = note: Snippet: 178
+  = note: Debug: $DIR/print_span.rs:9:21 - 9:30
+  = note: Snippet: Some(21)?
+  = note: Source: File(
+              FileInfo {
```

Comment:
Maybe for `FileInfo` the debug representation could be just the file path? This would reduce the snapshots removing the noisy `file:, span_src: SpanSrcId(..)` portions.

Reply:
IDK, with rustc Span I often find it annoying, that they always print the path, since I often wanted to know the exact values they store. Reworking the `Debug` implementations of Marker structs to only include "public" information should be on the todo list, but not in this branch IMO

Reply:
Beware that you can use the `{:#?}` syntax to print an alterante debug representation of your type. It isn't very flexible (it's just a boolean), but it allows you to customize the debug representation at least somehow.

This way if you use a terse form `{:?}` the file info could print just the file path, and if you use the `{:#?}` then the debug impl could include everything (the verbose debug output).

To implement this you need to use the [`alternate` method of the formatter](https://doc.rust-lang.org/stable/std/fmt/struct.Formatter.html#method.alternate)

Reply:
Spans will most often be printed as part of other nodes. Therefore, I feel like the alternative print is not the best solution. I would like the API to this file output, whenever it would print `SpanId(...)`. That's a relatively simple change, but something for the future :)